frontend/javascripts/dashboard/dataset/dataset_cache_provider.js

```diff
@@ -155,9 +171,48 @@ export default function DatasetCacheProvider({ children }: { children: Node }) {
     }
   }
 
+  async function updateCachedDataset(dataset: APIDataset) {
+    setIsLoading(true);
+
+    const updatedDatasets = datasets.map(currentDataset => {
+      if (
+        dataset.name === currentDataset.name &&
+        dataset.owningOrganization === currentDataset.owningOrganization
+      ) {
+        return dataset;
+      } else {
+        return currentDataset;
+      }
+    });
+    setDatasets(updatedDatasets);
+    datasetCache.set(updatedDatasets);
+    try {
+      const previousDatasetUpdatePromise = pendingDatasetUpdates[dataset.name];
+      const newDatasetUpdatePromise = Promise.resolve(previousDatasetUpdatePromise).then(async () =>
```

Comment:
While the behavior seems to be correct, I stumbled upon this, because it confused me that no null-check is done and I personally only use `Promise.resolve` to create a fulfilled Promise with a dedicated value. I would either add a comment which explains why the code works or rewrite it like this:

```suggestion
      const previousDatasetUpdatePromise = pendingDatasetUpdates[dataset.name] || Promise.resolve();
      const newDatasetUpdatePromise = previousDatasetUpdatePromise.then(() =>
```

Also note that I removed the `async` keyword as it wasn't necessary.

Reply:
I am fine with your version :)

I just learned about this behaviour of `Promise.resolve` while reading on how to create a "promise-queue" :)